### PR TITLE
Update methods that check for widget presence

### PR DESCRIPTION
**Pull request by adsteel**
Addressing issue https://github.com/mojotech/dill/issues/83

### CODE REVIEW

**Comment by adsteel** [comment]
Current plan is to:
- Formally deprecate `widget?`, `present?`, `absent?`
- Replace with `visible?` `not_visible?`
- Update the Readme

**Comment by adsteel** [comment]
@david What do you think of this? I think maybe adding a `widget.visible?` method will be more complex, since we'll have to deal with the error raised for unfound widgets.

**Comment by david** [comment]
@adsteel I think `not_visible?` will force people to wait, since it`s using `visible?` under the covers, which does wait.

**Comment by adsteel** [comment]
@david Good point. I've switched it to use `has_no_selector`, which waits for the element to not be on the page. I've also added a second commit that swaps the deprecated methods for the new methods in tests.

File: lib/dill.rb [hunk 1]
```diff
@@ -20,4 +20,10 @@ class Missing < StandardError; end
   class MissingWidget < StandardError; end
   class AmbiguousWidget < StandardError; end
   class InvalidOption < StandardError; end
+
+  def deprecate(method, alternate_method, once=false)
```
Comment (aesnyder) [inline lib/dill.rb]: nice

File: lib/dill/widgets/parts/container.rb [hunk 2]
```diff
@@ -1,12 +1,23 @@
 module Dill
   module WidgetParts
     module Container
+      include Dill
```
Comment (aesnyder) [inline lib/dill/widgets/parts/container.rb]: why the need to reinclude dill?

File: lib/dill/widgets/parts/container.rb [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
+
       def has_widget?(name, *args)
+        deprecate('has_widget? and its alias widget?', 'visible?')
```
Comment (aesnyder) [inline lib/dill/widgets/parts/container.rb]: Is that to get access to the deprecate method here?

File: lib/dill/widgets/parts/container.rb (hunk 2 above)
Comment (adsteel) [inline lib/dill/widgets/parts/container.rb]: To allow access to the deprecated method.

**Comment by aesnyder** [comment]
:+1: LGTM

**Pull request merged**